amplipi/streams.py

```diff
@@ -1074,18 +1103,18 @@ def connect(self, src):
 
     fmradio_args = [sys.executable, f"{utils.get_folder('streams')}/fmradio.py", self.freq, utils.real_output_device(src), '--song-info', song_info_path, '--log', log_file_path]
     print(f'running: {fmradio_args}')
-    self.proc = subprocess.Popen(args=fmradio_args, stdin=subprocess.PIPE, stdout=subprocess.PIPE, stderr=subprocess.PIPE, preexec_fn=os.setpgrp)
+    self.metadata_proc = subprocess.Popen(args=fmradio_args, stdin=subprocess.PIPE, stdout=subprocess.PIPE, stderr=subprocess.PIPE, preexec_fn=os.setpgrp)
     self._connect(src)
 
   def _is_running(self):
-    if self.proc:
-      return self.proc.poll() is None
+    if self.metadata_proc:
+      return self.metadata_proc.poll() is None
     return False
 
   def disconnect(self):
     if self._is_running():
-      os.killpg(os.getpgid(self.proc.pid), signal.SIGKILL)
-    self.proc = None
+      os.killpg(os.getpgid(self.metadata_proc.pid), signal.SIGKILL)
```

Comment:
Should we try to send something over the fifo to exit cleanly? 

It looks like sending a newline would cleanly exit the command handler thread.  We could add something at the bottom of the metadata script that waits for the command thread to exit and then cases the metadata reader thread to exit as well.